pages/article/ArticleHandler.inc.php

```diff
@@ -71,9 +96,35 @@ function initialize($request, $args) {
 			$article = $articleDao->getById((int) $articleId, $journal->getId(), true);
 			$this->article = $article;
 		}
+	}
 
-		$galleyDao = DAORegistry::getDAO('ArticleGalleyDAO');
-		$this->galley = $galleyDao->getByBestGalleyId($galleyId, $this->article->getId());
+	/**
+	 * Handle article versions. Calls view().
+	 * @param $args array
+	 * @param $request Request
+	 */
+	function version($args, $request){
+		$articleId = $args[0];
+		$this->submissionRevision = $args[1];
+		$galleyId = isset($args[2]) ? $args[2] : 0;
+		array_splice($args, 1, 1);
+
+		// check if submission revision exists
+		$submissionDao = Application::getSubmissionDAO();
+		$submission = $submissionDao->getById($articleId, null, false, $this->submissionRevision);
```

Comment:
Hmm, the problem is, that the version number can only be read at the version function. So the function at initialize can only get the default values (the latest version) but for the display of the version specific data you need to call getPublishedArticleByBestArticleId again at version().